Clarify the status of all implementations (#457)
transparency-dev/tesseract: README.md

using the [Tessera](https://github.com/transparency-dev/tessera)
library to store data, and is aimed at running production-grade CT logs.

> At the moment, TesseraCT can run on GCP, AWS, a POSIX filesystems, or on some

Reviewer: ```suggestion
At the moment, TesseraCT can run on GCP, AWS, POSIX-compliant filesystems, or
```

> S3-compatible systems alongside a SQL database with [different levels of maturity](#mega-status).

Reviewer: ```suggestion
S3-compatible systems alongside a MySQL database with [different levels of maturity](#mega-status).
```

TesseraCT is under active development, and will reach alpha in 2025Q3 🚀.

|Platform         |Architecture            |Our use-case          |Performance|Binary                                  |Deployment                                                      |
|-----------------|------------------------|----------------------|-----------|----------------------------------------|----------------------------------------------------------------|

> |GCP              |Spanner + GCS + MIG     |public staging logs   |TODO       |[gcp](/cmd/tesseract/gcp/main.go)       |[doc](/deployment/live/gcp/static-ct-staging/logs/arche2025h1/) |

Reviewer 2: This is the first time `MIG` appears in the README. I would suggest adding a link to explain what it is.

https://cloud.google.com/compute/docs/instance-groups#managed_instance_groups

Author: I linked everything

|GCP              |Spanner + GCS + CloudRun|continuous integration|TODO       |[gcp](/cmd/tesseract/gcp/main.go)       |[example](/deployment/live/gcp/static-ct/logs/ci)               |
|GCP              |Spanner + GCS + VM      |codelab, load tests   |TODO       |[gcp](/cmd/tesseract/gcp/main.go)       |[doc](/deployment/live/gcp/test/)                               |
|AWS              |RDS + S3 + ECS          |continuous integration|TODO       |[aws](/cmd/tesseract/aws/main.go)       |[example](/deployment/live/aws/conformance/ci/)                 |
|AWS              |RDS + S3 + VM           |codelab, load tests   |TODO       |[aws](/cmd/tesseract/aws/main.go)       |[doc](/deployment/live/aws/test/)                               |

> |POSIX            |ZFS + VM                |codelab               |NA         |[posix](/cmd/experimental/posix/main.go)|[doc](/cmd/posix/)                                              |

Reviewer: This can now be updated to mention `continuous integration` too, since #467 has gone in :)

|-----------------|------------------------|----------------------|-----------|----------------------------------------|----------------------------------------------------------------|
|GCP              |Spanner + GCS + MIG     |public staging logs   |TODO       |[gcp](/cmd/tesseract/gcp/main.go)       |[doc](/deployment/live/gcp/static-ct-staging/logs/arche2025h1/) |
|GCP              |Spanner + GCS + CloudRun|continuous integration|TODO       |[gcp](/cmd/tesseract/gcp/main.go)       |[example](/deployment/live/gcp/static-ct/logs/ci)               |
|GCP              |Spanner + GCS + VM      |codelab, load tests   |TODO       |[gcp](/cmd/tesseract/gcp/main.go)       |[doc](/deployment/live/gcp/test/)                               |
|AWS              |RDS + S3 + ECS          |continuous integration|TODO       |[aws](/cmd/tesseract/aws/main.go)       |[example](/deployment/live/aws/conformance/ci/)                 |
|AWS              |RDS + S3 + VM           |codelab, load tests   |TODO       |[aws](/cmd/tesseract/aws/main.go)       |[doc](/deployment/live/aws/test/)                               |

> |POSIX            |ZFS + VM                |one-off test          |NA         |[posix](/cmd/experimental/posix/main.go)|[doc](/cmd/experimental/posix/)                                 |

Reviewer: POSIX has the equivalent `codelab` in the README.

I'll add CI coverage and load-test info and we can update this row with those later.

|Custom S3 + MySQL|MinIO + MySQL + VM      |one-off test          |NA         |[aws](/cmd/tesseract/aws/main.go)       |[doc](/docs/architecture/NONCLOUD.md)                           |

These deployments come with different levels of maturity depending on
our use-case.

> We dedicated most of our time to the GCP with Spanner + GCS + MIG implementation

Reviewer: How about:

```suggestion
Our primary focus so far has been on the GCP with Spanner + GCS + MIG configuration
since we use it for our [public staging logs](#test_tube-public-test-instances). However we believe all implementations are correct, and we'd love to [hear your feedback](#wave-contact) on any of these implementations.

```

     - Architecture
       - GCP: TODO
       - AWS: TODO
       - POSIX: TODO

>        - [S3+MySQL](./docs/architecture/NONCLOUD.md)

Author: Yeah.... I think we might delete this file altogether, still waiting to see what we do exactly with SSSQL 👀

### Why these platforms?

After chatting with various CT log operators, we decided to focus on GCP, AWS,

> and to explore non-cloud-native deloyments. We welcome feedbacks on these and

Reviewer: ```suggestion
and to explore non-cloud-native deloyments. We welcome feedback on these and
```
